libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendTaskModuleCardResponse.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @module botbuilder-dialogs-adaptive-teams
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BoolExpression, BoolExpressionConverter } from 'adaptive-expressions';
+import { Activity, Attachment, TaskModuleContinueResponse, TaskModuleResponse, TaskModuleTaskInfo } from 'botbuilder';
+import {
+    Converter,
+    ConverterFactory,
+    DialogConfiguration,
+    DialogContext,
+    DialogStateManager,
+    DialogTurnResult,
+    TemplateInterface,
+} from 'botbuilder-dialogs';
+import { ActivityTemplateConverter } from 'botbuilder-dialogs-adaptive/lib/converters';
+import { BaseSendTaskModuleContinueResponse } from './baseSendTaskModuleContinueResponse';
+import { BaseTeamsCacheInfoResponseDialog } from './baseTeamsCacheInfoResponseDialog';
+
+export interface SendTaskModuleCardResponseConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    card?: TemplateInterface<Activity, DialogStateManager>;
+}
+
+/**
+ * Send a messaging extension 'message' response.
+ */
+export class SendTaskModuleCardResponse
+    extends BaseSendTaskModuleContinueResponse
+    implements SendTaskModuleCardResponseConfiguration {
+    /**
+     * Class identifier.
+     */
+    public static $kind = 'Teams.SendTaskModuleCardResponse';
+
+    /**
+     * Gets or sets template for the attachment template of a Thumbnail or Hero Card to send.
+     */
+    public card: TemplateInterface<Activity, DialogStateManager>;
+
+    public getConverter(property: keyof SendTaskModuleCardResponseConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'card':
+                return new ActivityTemplateConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled?.getValue(dc.state)) {
+            return dc.endDialog();
+        }
+
+        let attachment;
```

Comment:
This should be `let attachment: Attachment;` and the `as Attachment` below should be removed.